src/components/canrawview/canrawview_p.h

```diff
@@ -97,7 +108,11 @@ private slots:
     void sort(const int currentIndex, QList<int> &clickedIndexes)
     {
         clickedIndexes.removeFirst();
-        clickedIndexes.append(currentIndex);
+
+        if (currentIndex == 2)
+            clickedIndexes.append(1);
+        else
+            clickedIndexes.append(currentIndex);
 
         if ((clickedIndexes[2] == clickedIndexes[1]) && (clickedIndexes[1] == clickedIndexes[0]))
```

Comment:
I believe you can achieve much simpler sorting logic by mixture of sortIndicatorOrder and sortIndicatorSection methods.